str_test.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env nash
+
+# testing forr str.sh
+
+import ./str
+
+fn testjoin(expected, lst, sep) {
+	got <= join($lst, $sep)
+
+	if $got != $expected {
+		printf "Failed to join: '%s' != '%s'" $got $expected
+
+		return "1"
+	}
+
+	return "0"
+}
+
+# testJoin returns 0 in case of tests pass and 1 otherwise
+fn TestJoin() {
+	tests = (
+		(
+			"scratching my itches"
+			(scratching my itches)
+			" "
+		)
+		(
+			"plan9/from/outer/space"
+			(plan9 from outer space)
+			"/"
+		)
+		(
+			"writing tests in shellscript we are making history"
+			(writing tests in shellscript we are making history)
+			" "
+		)
+	)
```

Comment:
Well, I thought about testing exactly that the program would abort properly and not accept the parameter and return some crazy result. But I think you are right, this should be something for the interpreter (nash) to worry about and add as test case.